services/brig/test/integration/API/Provider.hs

```diff
@@ -1601,3 +1793,98 @@ addBotConv brig cannon uid1 uid2 cid pid sid buf =
+    -- original name, only with the new name
+    let _upd = emptyUpdateService { updateServiceName = Just newName }
+    updateService brig pid sid _upd !!! const 200 === statusCode
+    searchFor "after name change" origName []
+    searchFor "after name change" newName [(sid, newName)]
+    -- Let's rollback; now searching for the new name should return nothing
+    let _upd = emptyUpdateService { updateServiceName = Just origName }
+    updateService brig pid sid _upd !!! const 200 === statusCode
+    searchFor "after rollback" newName []
+    searchFor "after rollback" origName [(sid, origName)]
+  where
+    newName = Name (uniq <> "|NewName")
+    searchFor testName qry expected =
+        search qry >>=
+        assertServiceDetails (testName <> ": searching for " <> show qry) expected
+    emptyUpdateService = UpdateService
+        { updateServiceName    = Nothing
+        , updateServiceSummary = Nothing
+        , updateServiceDescr   = Nothing
+        , updateServiceAssets  = Nothing
+        , updateServiceTags    = Nothing
+        }
+
+-- | Check that lists match and there are no results on the second page.
+assertServiceDetails
+    :: (HasCallStack, MonadIO m)
+    => String -> [(ServiceId, Name)] -> ServiceProfilePage -> m ()
+assertServiceDetails testName expected page = liftIO $ do
+    let ids   = map serviceProfileId   (serviceProfilePageResults page)
+    let names = map serviceProfileName (serviceProfilePageResults page)
+    assertEqual (testName <> ": names") (map (fromName . snd) expected) (map fromName names)
+    assertEqual (testName <> ": ids") (map fst expected) ids
+    -- This is commented out because otherwise tests wouldn't pass
+    -- (even though they should!). See Note [buggy pagination] for more
+    -- details.
+    --
+    -- assertEqual (testName <> ": no hidden results") False (serviceProfilePageHasMore page)
+
+-- | Call the endpoint that searches through all services.
+searchServices
+    :: HasCallStack
+    => Brig -> Int -> UserId -> Maybe Name -> Maybe MatchAny -> Http ServiceProfilePage
+searchServices brig size uid mbStart mbTags = case (mbStart, mbTags) of
+    (Nothing, Nothing) ->
+        error "searchServices: query not supported"
+    (Just start, Nothing) ->
+        fromMaybe (error "searchServices") . decodeBody <$>
+            (listServiceProfilesByPrefix brig uid start size
+             <!! const 200 === statusCode)
+    (_, Just tags) ->
+        fromMaybe (error "searchServices") . decodeBody <$>
+            (listServiceProfilesByTag brig uid tags mbStart size
+             <!! const 200 === statusCode)
+
+-- | Call the endpoint that searches through whitelisted services.
+searchServiceWhitelist
+    :: HasCallStack
+    => Brig -> Int -> UserId -> TeamId -> Maybe Text -> Http ServiceProfilePage
+searchServiceWhitelist brig size uid tid mbStart =
+    fromMaybe (error "searchServiceWhitelist") . decodeBody <$>
```

Comment:
if you use `either (error . show) id` here, you will get more information from `decodeBody` in case of error.  (if this is the one from `/services/brig/src/Brig/RPC.hs` anyway.  if it's not and thsi won't work, `decodeBody` should be fixed.)